src/service/template-impl.js

```diff
@@ -53,7 +54,7 @@ export class BaseTemplate {
     this.win = element.ownerDocument.defaultView || win;
 
     /** @private @const */
-    this.viewer_ = getServiceForDoc(this.element, 'viewer');
+    this.viewer_ = Services.viewerForDoc(this.element);
```

Comment:
Do we know that `Services` never internally uses `getServiceForDoc` in the static methods? I don't see how banning in the extension files themselves helps, without changing every extension file to use async services.

Reply:
Good question. IMO less likely that `Services` methods will be wrong since they simply won't work consistently. E.g. I audited already and only one method is problematic (see the TODO) vs. many instances in extension code. I also considered splitting `Services` into two so we could enforce this more granularly, but first things first.

> I don't see how banning in the extension files themselves helps, without changing every extension file to use async services.

Yea, that's next. This stops the bleeding. 

> Next is to start migrating extension code to use the async API.